magefiles/utils.go

```diff
@@ -150,3 +165,120 @@ func renderTemplate(destination, templatePath string, templateData interface{},
 
 	return nil
 }
+
+// helper function used for testing jvm-build-service PRs
+func overridePipelineBundle() error {
+	const buildTemplatesBundleKustomizeUrl = "https://raw.githubusercontent.com/redhat-appstudio/infra-deployments/main/components/build-templates/kustomization.yaml"
+	const bundleName = "hacbs_build_bundle"
+	var newReqprocessorImage = os.Getenv("JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE")
+
+	const imagePushRepo = "quay.io/redhat-appstudio-qe/test-images"
+	prSHA := openshiftJobSpec.Refs.Pulls[0].SHA
+	var newS2iJavaTaskRef, _ = name.ParseReference(fmt.Sprintf("%s:task-bundle-%s", imagePushRepo, prSHA))
+	var newJavaBuilderPipelineRef, _ = name.ParseReference(fmt.Sprintf("%s:pipeline-bundle-%s", imagePushRepo, prSHA))
+
+	var rawRegistryCreds []byte
+	var err error
+	var homeDir string
+
+	var newTaskYaml, newPipelineYaml []byte
+
+	registryCreds := os.Getenv("QUAY_TOKEN")
```

Comment:
nit: It feels like this whole section of code around the docker.cofngi could be factored out into a generic util function for creating/updating the docker.config